### PR TITLE
Tidy/minor code changes

### PR DESCRIPTION
**Pull request by i10416**
This commit changes some codes that I found are a bit verbose and complex without changing their logic.

### CODE REVIEW

File: gcloud-sdk/src/token_source/ext_creds_source/mod.rs [hunk 1]
```diff
@@ -69,41 +69,33 @@ pub async fn subject_token_url(
...
-                ExternalCredentialUrlFormat::Json(json_settings) => {
-                    let json: serde_json::Value = response.json().await?;
-                    let json_object = json.as_object().ok_or_else(|| {
-                        crate::error::ErrorKind::ExternalCredsSourceError(format!(
-                            "External subject JSON format is not object: {}",
-                            json
-                        ))
-                    })?;
+        match &url_creds.format {
+            None | Some(ExternalCredentialUrlFormat::Text) => Ok(response.text().await?.into()),
```
Comment (i10416) [inline gcloud-sdk/src/token_source/ext_creds_source/mod.rs:73]: We can merge `None` case and `Some(Text)` case in one clause.

File: gcloud-sdk/src/token_source/ext_creds_source/mod.rs [hunk 2]
```diff
@@ -129,47 +121,37 @@ pub async fn subject_token_file(
...
-                let json: serde_json::Value =
-                    serde_json::from_str(file_content.as_str()).map_err(|e| {
-                        crate::error::ErrorKind::ExternalCredsSourceError(format!(
-                            "External file JSON format error: {}",
-                            e
-                        ))
-                    })?;
-                let json_object = json.as_object().ok_or_else(|| {
+    match &url_creds.format {
+        None | Some(ExternalCredentialUrlFormat::Text) => Ok(file_content.into()),
```
Comment (i10416) [inline gcloud-sdk/src/token_source/ext_creds_source/mod.rs:125]: same as https://github.com/abdolence/gcloud-sdk-rs/pull/118/files#r1441784227

File: gcloud-sdk/src/token_source/credentials.rs [hunk 3]
```diff
@@ -18,6 +18,16 @@ pub enum Credentials {
     ExternalAccount(ExternalAccount),
 }
 
+impl Credentials {
+    pub(crate) fn quota_project_id(&self) -> Option<&str> {
```
Comment (i10416) [inline gcloud-sdk/src/token_source/credentials.rs:22]: I usually put this kind of field access  in struct impl section.

File: gcloud-sdk/src/token_source/credentials.rs [hunk 4]
```diff
@@ -166,7 +176,7 @@ mod jwt {
         }
     }
 
-    #[derive(Debug, serde::Serialize, serde::Deserialize)]
```
Comment (i10416) [inline gcloud-sdk/src/token_source/credentials.rs:169]: No need to derive Deserialize.

**Review by abdolence — APPROVED** [review]
Thanks for clean ups! 
There is still some old verbose code around token sources particularly that I clean from time to time myself.

**Pull request merged**